src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -37,17 +37,26 @@
 #include <shogun/mathematics/linalg/GPUMemoryBase.h>
 #include <memory>
 
-#ifndef Linalg_Backend_Base_H__
-#define Linalg_Backend_Base_H__
+#ifndef LINALG_BACKEND_BASE_H__
+#define LINALG_BACKEND_BASE_H__
 
 namespace shogun
 {
 
-/** Base class for generic linalg methods */
+/** @brief Base interface of generic linalg methods
+ * and generic memory transfer methods.
+ */
 class LinalgBackendBase
 {
 public:
-	/** Generic vector dot-product macro */
+	/**
+	 * Wrapper method of vector dot-product that works with generic vectors.
```

Comment:
As said, please reference the method in the linalg main user interface here.
No need to re-write all the `@param` and `@return` here, we only want to write this down once. Only comments that dev specific should go here. Like "Override in backend specializations to support this operation"